### PR TITLE
SBS-4017: add support for coregis feature endpoints, including search

### PR DESCRIPTION
**Pull request by vincent-cognite**

### CODE REVIEW

File: tests/tests_integration/test_api/test_geospatial.py [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
-from cognite.experimental.data_classes.geospatial import FeatureType
+from cognite.experimental.data_classes.geospatial import Feature, FeatureType
 
 COGNITE_CLIENT = CogniteClient()
 COGNITE_DISABLE_GZIP = "COGNITE_DISABLE_GZIP"
 
 
 @pytest.fixture
 def new_feature_type():
+    external_id = "FT_" + uuid.uuid4().hex[:10]
```
Comment (pawelzwolenskicognite) [inline tests/tests_integration/test_api/test_geospatial.py:15]: the only thing I know about python is that you can use string templates :)
https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python

File: cognite/experimental/_api/geospatial.py [hunk 2]
```diff
@@ -101,3 +95,137 @@ def retrieve_feature_types(self, external_id: Union[str, List[str]] = None) -> F
...
+        Args:
+            feature_type : Feature type definition for the features to create.
+            feature: one feature or a list of features to create
+
+        Returns:
+            Union[Feature, FeatureList]: Created features
+
+        Examples:
+
+            Create a new feature::
```
Comment (pawelzwolenskicognite) [inline cognite/experimental/_api/geospatial.py:114]: :: -> :

File: tests/tests_integration/test_api/test_geospatial.py [hunk 3]
```diff
@@ -30,10 +42,34 @@ def disable_gzip():
...
+    def test_update_features(self, new_feature_type, new_feature):
+        res = COGNITE_CLIENT.geospatial.update_features(
+            feature_type=new_feature_type, feature=Feature(external_id=new_feature.external_id, temperature=6.237)
+        )
+        assert res.external_id == new_feature.external_id
+        assert res.temperature == 6.237
+
+    def test_search_features(self, new_feature_type, new_feature):
+        res = COGNITE_CLIENT.geospatial.search_features(
+            feature_type=new_feature_type, filter={"range": {"attribute": "temperature", "gt": 12.0}}, limit=10
```
Comment (pawelzwolenskicognite) [inline tests/tests_integration/test_api/test_geospatial.py:68]: that's nice that we can use python dictionaries for filters!

**Review by pawelzwolenskicognite — APPROVED** [review]
looks good!

File: tests/tests_integration/test_api/test_geospatial.py (hunk 1 above)
Comment (vincent-cognite) [reply tests/tests_integration/test_api/test_geospatial.py:15]: which language does not have that nowadays?
let's use string templates then!

Comment (pawelzwolenskicognite) [reply tests/tests_integration/test_api/test_geospatial.py]: java :P

**Pull request merged**